stl/inc/filesystem

```diff
@@ -1206,78 +1205,82 @@ namespace filesystem {
             // "3. Replace each directory-separator with a preferred-separator.
             // [ Note: The generic pathname grammar (29.11.7.1) defines directory-separator
             // as one or more slashes and preferred-separators. -end note ]"
-            list<wstring_view> _Lst; // Empty wstring_view means directory-separator
-                                     // that will be normalized to a preferred-separator.
-                                     // Non-empty wstring_view means filename.
-            for (auto _Next = _Root_name_end; _Next != _Last;) {
-                if (_Is_slash(*_Next)) {
-                    if (_Lst.empty() || !_Lst.back().empty()) {
+            vector<wstring_view> _Vec; // Empty wstring_view means directory-separator
+            // that will be normalized to a preferred-separator.
+            // Non-empty wstring_view means filename.
+            _Vec.reserve(13); // avoid frequent re-allocations

```

Comment:
this is vector's capacity after 10 pushes in current implementation.

Reply:
Hmm... it seems better to use something like `boost::small_vector` (no change requested).